app/src/redux-resources/analytics/hooks/useModuleAnalytics.ts

```diff
@@ -0,0 +1,154 @@
+    // Checks param type and returns variables found
+    if (isModuleOnlyParams(params)) {
+        return { moduleId: String(params.moduleId), celsius: "" };
+    }
+    if (isTemperatureParams(params)) {
+        return { moduleId: String(params.moduleId), celsius: String(params.celsius) };
+    }
+    return { moduleId: "", celsius: "" }; // Default return for unexpected cases
+}
+
+export function useModuleCommandAnalytics(modules?: AttachedModule[]): UseModuleCommandAnalyticsResult {
+    const doTrackEvent = useTrackEvent();
+    const reportModuleCommand = ({
+        runId,
+        moduleType,
+        action,
+        result,
+        errorDetails,
+        serialNumber,
+        temperature,
+        params,
+        firmwareVersion
+    }: ReportModuleActionParams & { errorDetails?: string }): void => {
+        if (typeof action !== 'string' || !commandType.includes(action)) {
+            return; // Exit early if action is invalid or not in the commandType array
+        }
+        // Ensure params is either ModuleOnlyParams or TemperatureParams before processing
+        if (typeof params != null){
+            if (!isModuleOnlyParams(params) && !isTemperatureParams(params)) {
+                console.warn("Unexpected params format:", params);
+                temperature = "UNKNOWN"
+                serialNumber = "UNKNOWN"
+            }
+            else if(typeof runId == 'string'){
+                // If runId is passed, find moduleId serialNumber match
+                const { data: deckConfig = [] } = useNotifyDeckConfigurationQuery({
+                    refetchInterval: DECK_CONFIG_POLL_MS,
+                })
+                const moduleQuery = useModulesQuery({
+                    refetchInterval: MODULE_REFETCH_INTERVAL_MS,
+                })
+                const attachedModules = moduleQuery?.data?.data ?? []
+                const mostRecentAnalysis = useMostRecentCompletedAnalysis(runId)
+                const protocolModulesInfo =
+                    mostRecentAnalysis != null
+                    ? getProtocolModulesInfo(mostRecentAnalysis, deckDef)
+                    : []
+                const attachedProtocolModuleMatches = getAttachedProtocolModuleMatches(
+                    attachedModules,
+                    protocolModulesInfo,
+                    deckConfig
+                )
+                const matchedModules = attachedProtocolModuleMatches.map(module=> ({
+                    moduleModel: module.attachedModuleMatch?.moduleType,
+                    moduleId: module.attachedModuleMatch?.id,
+                    serialNumber: module.attachedModuleMatch?.serialNumber,
+                    firmwareVersion: module.attachedModuleMatch?.firmwareVersion
+                }))
+                // Extract moduleId and temp
+                const { moduleId, celsius } = isParamType(params);
+                // update temperature
+                temperature = celsius !== "" ? celsius : temperature;
```

Comment:
```suggestion
               const reportedTemperature = celsius !== "" ? celsius : temperature;
```

Avoid mutating state. This can lead to some really nasty bugs.